src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -9,45 +9,86 @@ import spray.json.DefaultJsonProtocol._
+        case _ => throw new DeserializationException("illegal array type")
+      }
+    case _ => throw new DeserializationException("unexpected json type")
+  }
+}
 
-  implicit val impSubmissionStats = jsonFormat3(SubmissionStats)
-  implicit val impRawlsWorkspace = jsonFormat11(RawlsWorkspace)
-  implicit val impRawlsWorkspaceResponse = jsonFormat4(RawlsWorkspaceResponse)
-  implicit val impUIWorkspace = jsonFormat12(UIWorkspace)
-  implicit val impUIWorkspaceResponse = jsonFormat4(UIWorkspaceResponse)
+//Serializes attribute lists to e.g. { "itemsType" : "AttributeValue", "items" : [1,2,3] }
+trait TypedAttributeListSerializer extends AttributeListSerializer {
+  val LIST_ITEMS_TYPE_KEY = "itemsType"
+  val LIST_ITEMS_KEY = "items"
+  val LIST_OBJECT_KEYS = Set(LIST_ITEMS_TYPE_KEY, LIST_ITEMS_KEY)
+
+  val VALUE_LIST_TYPE = "AttributeValue"
+  val REF_LIST_TYPE = "EntityReference"
+
+  def writeListType(obj: Attribute): JsValue = obj match {
+    //lists
+    case AttributeValueEmptyList => writeAttributeList(VALUE_LIST_TYPE, Seq.empty[AttributeValue])
+    case AttributeValueList(l) => writeAttributeList(VALUE_LIST_TYPE, l)
+    case AttributeEntityReferenceEmptyList => writeAttributeList(REF_LIST_TYPE, Seq.empty[AttributeEntityReference])
+    case AttributeEntityReferenceList(l) => writeAttributeList(REF_LIST_TYPE, l)
+  }
 
-  implicit val impEntity = jsonFormat5(Entity)
-  implicit val impEntityCreateResult = jsonFormat4(EntityCreateResult)
-  implicit val impEntityWithType = jsonFormat3(EntityWithType)
-  implicit val impEntityCopyDefinition = jsonFormat3(EntityCopyDefinition)
-  implicit val impEntityCopyWithDestinationDefinition = jsonFormat4(EntityCopyWithDestinationDefinition)
-  implicit val impEntityId = jsonFormat2(EntityId)
-  implicit val impEntityDelete = jsonFormat2(EntityDeleteDefinition)
+  def readListType(json: JsValue): Attribute = json match {
+    case JsObject(members) if LIST_OBJECT_KEYS subsetOf members.keySet => readAttributeList(members)
 
-  implicit val impMethodConfiguration = jsonFormat8(MethodConfiguration)
-  implicit val impMethodConfigurationRename = jsonFormat3(MethodConfigurationRename)
+    case _ => throw new DeserializationException("unexpected json type")
+  }
 
-  implicit val impDestination = jsonFormat3(MethodConfigurationId)
-  implicit val impMethodConfigurationCopy = jsonFormat4(MethodConfigurationCopy)
-  implicit val impConfigurationCopyIngest = jsonFormat5(CopyConfigurationIngest)
-  implicit val impMethodConfigurationPublish = jsonFormat3(MethodConfigurationPublish)
-  implicit val impPublishConfigurationIngest = jsonFormat4(PublishConfigurationIngest)
+  def writeAttributeList[T <: Attribute](listType: String, list: Seq[T]): JsValue = {
+    JsObject( Map(LIST_ITEMS_TYPE_KEY -> JsString(listType), LIST_ITEMS_KEY -> JsArray(list.map( writeAttribute ).toSeq:_*)) )
+  }
```

Comment:
should this verify that `listType` is one of `VALUE_LIST_TYPE` or `REF_LIST_TYPE`?